api/task_dag.go

```diff
@@ -0,0 +1,30 @@
+package api
+
+// TaskDAG describes task dependency relationship.
+// FromTaskID is blocked by ToTaskID
+// It's rather DAGEdge than DAG
+type TaskDAG struct {
+	ID int
+
+	// Standard fields
+	CreatedTs int64
+	UpdatedTs int64
+
+	// Domain Specific fields
+	FromTaskID int
+	ToTaskID   int
+	Payload    string
+}
+
+// TaskDAGCreate is the API message to create TaskDAG.
+type TaskDAGCreate struct {
+	FromTaskID int
+	ToTaskID   int
+	Payload    string
+}
+
+// TaskDAGFind is the API message to find TaskDAG.
+type TaskDAGFind struct {
+	FromTaskID *int
+	ToTaskID   *int
```

Comment:
nit: we can keep ToTaskID only.

https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it

Reply:
Keep FromTaskID only now. PTAL